nntrainer/layers/layer_node.cpp

```diff
@@ -362,7 +362,7 @@ void LayerNode::read(std::ifstream &file) {
     << __func__ << " layer needs to be finalized first!";
   for (unsigned int i = 0; i < run_context->getNumWeights(); ++i) {
     /// @note dependent weights are only be read at the source
-    if (run_context->isWeightDependent(i)) {
+    if (!run_context->isGradientLastAccess(i)) {
```

Comment:
isGradientFirstAccess(i)?

Reply:
This is correct, gradientLastAccess == weightFirstAccess.

This deserves better naming, but for now it is being untouched with additional comment here as a quickfix

Reply:
ok.